drivers/utils/driver_utils.bal

```diff
@@ -46,22 +58,24 @@ public function initializeDriverHttpClients(string? hubUrl, string? paymentNetwo
     }
 }
 
+# Start scheduled job for initializing http clients to communicate with destination drivers.
+# + return - error
 public function initializeDestinationDriverClients() returns error? {
 
-    // ToDo: This should be called by a periodic scheduler
-    Metadata[] metadataList = check getdriverMetadataFromHub();
-
-    foreach var metadata in metadataList {
-        http:Client driverHttpClient = check new (metadata.paymentEndpoint);
-        // Add the client to the map with countryCode as the key
-        httpClientMap[metadata.countryCode] = driverHttpClient;
-        log:printInfo("Http client for the destination " + metadata.countryCode + " created");
-    }
+    //ToDo : Make the periodic interval configurable
+    task:JobId id = check task:scheduleJobRecurByFrequency(new DestinationClientInitializationJob(), 60);
+    log:printInfo("Started DestinationClientInitializationJob with job id: " + id.toString());
 }
 
+# Register driver at payments hub.
+#
+# + driverName - name of the driver  
+# + countryCode - country code of the driver  
+# + paymentEndpoint - payments endpoint which can be called by source drivers
+# + return - error
 public function registerDriverAtHub(string driverName, string countryCode, string paymentEndpoint) returns error? {
 
-    log:printInfo("Registering driver " + driverName + " at payments hub."); // todo - do we need paymentEndpoint??
+    log:printInfo("Registering driver " + driverName + " at payments hub.");
```

Comment:
Shall we make the `driver` a record and log the driver.toString()?

Reply:
Fixed by https://github.com/digital-payments-hub/digital-payments-hub/pull/10/commits/391157a9c752e66dbb67c95c67ccbc8b51bb6254